### PR TITLE
Chimera options

### PR DESCRIPTION
**Pull request by benjjneb**
This pull request addresses #50 Two options were added.

`chimera-method`: valid options are `none`, `pooled` and `consensus` (where `pooled` is the current behavior).

`min-parent-abundance`: A numeric value which should be 1.0 or greater.

I've added the options to the R scripts and verified thy work. I also added in python code that passes the CI tests, but needs to be reviewed and probably fixed. The python side has not been tested. Possibly a test should be added as well?

### CODE REVIEW

File: q2_dada2/_denoise.py [hunk 1]
```diff
@@ -44,6 +44,10 @@ def _check_featureless_table(fp):
     'trim_left_r': _WHOLE_NUM,
     'max_ee': _NAT_NUM,
     'trunc_q': _WHOLE_NUM,
+    'chimera_method': _SKIP,
+    # TEST chimera_method FOR VALIDITY HERE?
+    # Value should be 'pooled', 'consensus' or 'none'
```
Comment (ebolyen) [inline q2_dada2/_denoise.py]: We can skip this if we add a predicate to the semantic type (I'll comment on that further down).

File: q2_dada2/plugin_setup.py [hunk 2]
```diff
@@ -41,6 +41,9 @@
                 'trim_left': qiime2.plugin.Int,
                 'max_ee': qiime2.plugin.Float,
                 'trunc_q': qiime2.plugin.Int,
+                'chimera_method': qiime2.plugin.Str,
+                # IS qiime2.plugin.Str THE RIGHT TYPE?
```
Comment (ebolyen) [inline q2_dada2/plugin_setup.py]: Yup!

This can be:
```python
qiime2.plugin.Str % qiime2.plugin.Choices({'pooled', 'consensus', 'none'})
```

We also now have better support for `qiime2.plugin.Range` for the other number primitives in the future.

File: q2_dada2/plugin_setup.py [hunk 3]
```diff
@@ -65,6 +68,20 @@
         'trunc_q': ('Reads are truncated at the first instance of a quality '
                     'score less than or equal to this value. If the resulting '
                     'read is then shorter than `trunc_len`, it is discarded.'),
+        'chimera_method': ('The method used to remove chimeras. '
+                           'none: No chimera removal is performed. '
+                           'pooled: All reads are pooled prior to chimera '
+                           'detection. consensus: Chimeras are detected in '
+                           'samples individually, and sequences found '
+                           'chimeric in a sufficient fraction of samples are '
+                           'removed.'),
```
Comment (ebolyen) [inline q2_dada2/plugin_setup.py]: I would quote the values like `"none"`, `"pooled"`, `"consensus"`.

File: q2_dada2/plugin_setup.py [hunk 4]
```diff
@@ -102,6 +119,9 @@
                 'trim_left_r': qiime2.plugin.Int,
                 'max_ee': qiime2.plugin.Float,
                 'trunc_q': qiime2.plugin.Int,
+                'chimera_method': qiime2.plugin.Str,
```
Comment (ebolyen) [inline q2_dada2/plugin_setup.py]: same as above

File: q2_dada2/_denoise.py (hunk 1 above)
Comment (ebolyen) [inline q2_dada2/_denoise.py]: If you wanted to validate independent of the semantic type, you could add: 
`'chimera_method': lambda x: x in {'pooled', 'consensus', 'none'},`.

But I think ideally this `_valid_inputs` dict would be removed entirely in the future (using `qiime2.plugin.Range`).

**Comment by gregcaporaso** [comment]
@benjjneb, think you'll be able to address @ebolyen's comments by Friday? We're hoping to have all new code in by Friday, so we can release next week. Let us know if you need any help with meeting that timeline.

**Comment by benjjneb** [comment]
A miracle occurred and travis accepted my white space!

I added in @ebolyen suggestions, but someone should probably take a quick look before merging this in.

**Comment by ebolyen** [comment]
I'm going to merge this and write some simple unit-tests exercising that the parameters are passed through. @benjjneb, if you have any specific edge-cases worth testing, I'd be happy to include them, otherwise it's just going to use our oversimple mock data.
